src/components/views/elements/ManageIntegsButton.js

```diff
@@ -0,0 +1,127 @@
+            this.scalarClient = new ScalarAuthClient();
+            this.scalarClient.connect().done(() => {
+                this.forceUpdate();
+            }, (err) => {
+                this.setState({ scalar_error: err});
+            });
+        }
+    }
+
+    componentWillUnmount() {
+        ScalarMessaging.stopListening();
+    }
+
+    onManageIntegrations(ev) {
+        ev.preventDefault();
+        const IntegrationsManager = sdk.getComponent("views.settings.IntegrationsManager");
+        Modal.createDialog(IntegrationsManager, {
+            src: (this.scalarClient !== null && this.scalarClient.hasCredentials()) ?
+                this.scalarClient.getScalarInterfaceUrlForRoom(this.props.room.roomId) :
+                null,
+            onFinished: ()=>{
+                if (this.props.onCancelClick) {
+                    this.props.onCancelClick(ev);
+                }
+            },
+        }, "mx_IntegrationsManager");
+    }
+
+    onShowIntegrationsError(ev) {
+        ev.preventDefault();
+        this.setState({
+            showIntegrationsError: !this.state.showIntegrationsError,
+        });
+    }
+
+    render() {
+        let integrationsButton;
+        let integrationsError;
+        if (this.scalarClient !== null) {
+            if (this.state.showIntegrationsError && this.state.scalar_error) {
+                console.error(this.state.scalar_error);
+                integrationsError = (
+                    <span className="mx_RoomSettings_integrationsButton_errorPopup">
+                        { _t('Could not connect to the integration server') }
+                    </span>
+                );
+            }
+
+            if (this.scalarClient.hasCredentials()) {
+                integrationsButton = (
+                    <AccessibleButton className="mx_RoomHeader_button" onClick={this.onManageIntegrations} title={ _t('Manage Integrations') }>
+                        <TintableSvg src="img/icons-apps.svg" width="35" height="35"/>
+                    </AccessibleButton>
+                );
+            } else if (this.state.scalar_error) {
+                integrationsButton = (
+                    <div className="mx_RoomSettings_integrationsButton_error" onClick={ this.onShowIntegrationsError }>
+                        <img src="img/warning.svg" title={_t('Integrations Error')} width="17"/>
+                        { integrationsError }
+                    </div>
+                );
+            } else {
+                integrationsButton = (
+                    <AccessibleButton className="mx_RoomHeader_button" onClick={this.onManageIntegrations} title={ _t('Manage Integrations') }>
+                        <TintableSvg src="img/icons-apps.svg" width="35" height="35"/>
+                    </AccessibleButton>
+                );
+            }
+        }
+
+        return integrationsButton;
+    }
+}
+
+ManageIntegsButton.propTypes = {
+    room: PropTypes.object.isRequired,
```

Comment:
I'd prefer a `roomId` going in here, `ManageIntegsButton` doesn't need the rest of room state.